code/integration-tests/runtime-tests/test/tests/pablo/constantProductDex/constantProductTests.ts

```diff
@@ -1,285 +1,1888 @@
+}
+
+describe("[SHORT] Pablo: Constant Product Test Suite", function () {
+  if (!pabloTestConfiguration.enabledTests.enabled) {
     console.log("Constant Product Tests are being skipped...");
     return;
   }
   this.timeout(3 * 60 * 1000);
   let api: ApiPromise;
-  let walletId1: KeyringPair, walletId2: KeyringPair, walletId3: KeyringPair, sudoKey: KeyringPair;
-  let poolId: number,
-    poolId2: number,
-    baseAssetId: number,
-    baseAsset2: number,
-    quoteAssetId: number,
-    falseQuoteAsset: number,
-    fee: number,
-    baseWeight: number;
-  let baseAmount: bigint, quoteAmount: bigint;
-  let transferredTokens: BN;
-  let walletId1Account: string, walletId2Account: string;
+  let sudoKey: KeyringPair,
+    poolOwnerWallet: KeyringPair,
+    walletLpProvider1: KeyringPair,
+    walletLpProvider2: KeyringPair,
+    walletLpProvider3: KeyringPair,
+    walletTrader1: KeyringPair;
+  let poolId1: number, poolId2: number, poolId3: number;
 
   before("Initialize variables", async function () {
     const { newClient, newKeyring } = await getNewConnection();
     api = newClient;
-    const { devWalletAlice, devWalletEve, devWalletFerdie } = getDevWallets(newKeyring);
+    const { devWalletAlice, devWalletFerdie } = getDevWallets(newKeyring);
+
     sudoKey = devWalletAlice;
-    walletId1 = devWalletEve.derive("/test/constantProductDex/walletId1");
-    walletId2 = devWalletFerdie.derive("/test/constantProductDex/walletId2");
-    walletId3 = devWalletAlice.derive("/test/constantProductDex/walletId3");
-    walletId1Account = api.createType("AccountId32", walletId1.address).toString();
-    walletId2Account = api.createType("AccountId32", walletId2.address).toString();
-    baseAssetId = 2;
-    quoteAssetId = 3;
-    baseAsset2 = 7;
-    falseQuoteAsset = 23;
-    baseAmount = Pica(250000);
-    quoteAmount = Pica(250000);
-    //sets the fee to 1.00%/Type Permill
-    fee = 10000;
-    //sets the weight of the asset pairs to 50.00%/Type Permill
-    baseWeight = 500000;
+    poolOwnerWallet = devWalletFerdie.derive("/test/pablo/pool/owner");
+    walletLpProvider1 = devWalletFerdie.derive("/test/pablo/lp/provider/1");
+    walletLpProvider2 = devWalletFerdie.derive("/test/pablo/lp/provider/2");
+    walletLpProvider3 = devWalletFerdie.derive("/test/pablo/lp/provider/3");
+    walletTrader1 = devWalletFerdie.derive("/test/pablo/trader/1");
   });
 
   before("Minting assets", async function () {
-    await mintAssetsToWallet(api, walletId1, sudoKey, [1, baseAssetId, quoteAssetId, baseAsset2, falseQuoteAsset]);
-    await mintAssetsToWallet(api, walletId2, sudoKey, [1, baseAssetId, quoteAssetId, baseAsset2, falseQuoteAsset]);
-    await mintAssetsToWallet(api, walletId3, sudoKey, [1]);
+    // ToDo: Enable following when pool creation works!
+    // await mintAssetsToWallet(api, poolOwnerWallet, sudoKey, [1]);
+    await mintAssetsToWallet(
+      api,
+      walletLpProvider1,
```

Comment:
An idea: we can pass the wallets as array and send this minting assets step in one go.

Reply:
Yes but we'd either need to update this method incl. all tests that call this method.
Or make the method more complex by differentiating the input.
Or create a second method that can handle multiple wallets.

And I'm not sure what would be the best option here.